packages/react/src/hooks/useIsMacOS.ts

```diff
@@ -0,0 +1,16 @@
+import {isMacOS as ssrUnsafeIsMacOS} from '@primer/behaviors/utils'
+import {useEffect, useState} from 'react'
+
+/**
+ * SSR-safe hook for determining if the current platform is MacOS. When rendering
+ * server-side, will default to non-MacOS and then re-render in an effect if the
+ * client turns out to be a MacOS device.
+ */
+export function useIsMacOS() {
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+  const [isMacOS, setIsMacOS] = useState(() => (window !== undefined ? ssrUnsafeIsMacOS() : false))
+
+  useEffect(() => setIsMacOS(ssrUnsafeIsMacOS()), [])
+
+  return isMacOS
+}
```

Comment:
In the current usage of this component internally, we call `isMacOS` directly at the top level of the file. This is obviously not good for SSR-compatibility, so I've introduced this new hook:

- In non-SSR, it causes no extra renders and returns `isMacOS` directly
- In SSR, it initially returns `false`, then causes a re-render after hydration if the value would change to `true` (if the client is using a Mac)

I exposed this hook publicly so that we can also use it elsewhere when actually binding keyboard shortcuts.